App/components/ProfileForm.js

```diff
@@ -83,89 +84,88 @@ class Register extends Component {
             />
             {props.touched.email && props.errors.email &&
               <Text style={{ fontSize: 10, color: 'red' }}>
-              {props.errors.email}</Text>}
+                {props.errors.email}</Text>}
 
             {!this.props.disableSid &&
               <TextInput
-              style={styles.input}
-              placeholder="Student Number"
-              onChangeText={props.handleChange('id')}
-              onBlur={() => props.setFieldTouched('id')}
-              value={props.values.id}
+                style={styles.input}
+                placeholder="Student Number"
+                onChangeText={props.handleChange('id')}
+                onBlur={() => props.setFieldTouched('id')}
+                value={props.values.id}
               />
             }
             {props.touched.id && props.errors.id &&
               <Text style={{ fontSize: 10, color: 'red' }}>
-              Valid Student ID required</Text>}
+                Valid Student ID required</Text>}
 
             <Text style={{ fontSize: 10 }}>Year of Study</Text>
             <Picker
               style={styles.text}
               selectedValue={this.state.yr}
               onValueChange={(itemValue, itemIndex) => {
                 props.setFieldValue('yr', itemValue)
-                this.setState({yr: itemValue})
+                this.setState({ yr: itemValue })
               }}>
-              <Picker.Item label={this.props.userData.yr} value={this.props.userData.yr}/>
```

Comment:
This caused a duplication of picker items which I think was causing the error. Use `selectedValue` (which is now properly set) to pick one of the actual picker items instead